### PR TITLE
fix expressions for masking layers

### PR DESCRIPTION
**Pull request by dzinad**
The issue is reproducible when you use different scaling expressions for primary and alternative routes. To catch these issues I added this setup to InactiveRouteLegStylingActivity in qa-test-app.
Before:
image
After:
image

### CODE REVIEW

File: libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt [hunk 1]
```diff
@@ -1646,22 +1646,11 @@ class MapboxRouteLineApi(
 
         val primaryRouteCasingExpressionProducer =
             RouteLineExpressionProvider {
-                if (routeLineOptions.styleInactiveRouteLegsIndependently) {
```
Comment (LukasPaczos) [inline libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt:1649]: Why does the base layer still need an independent code path for inactive leg styling and casing layer doesn't?

Comment (dzinad) [reply libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt:1649]: We don't have an "inactive leg casing colour" option. It was using travelled colour instead. But I now see that we use travelled casing colour in `updateTraveledRouteLine` for inactive legs. Is this expected?

Comment (LukasPaczos) [reply libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt:1649]: Could you help here @cafesilencio?

Comment (dzinad) [reply libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt:1649]: Let me revert this change and make it a separate PR. The width problems are fixed by MapboxRouteLineView changes.

Comment (dzinad) [reply libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt]: Here's the video of how the colours work on main @cafesilencio:

https://user-images.githubusercontent.com/32109537/235144208-ec3369c2-60fa-46c6-8969-690220bb2f28.mp4

File: changelog/unreleased/bugfixes/7132.md [hunk 2]
```diff
@@ -0,0 +1 @@
+- Fixed an issue where the route line casing of the active leg was too wide.
```
[the remark this answers is not in the document]
Comment (dzinad) [reply changelog/unreleased/bugfixes/7132.md:1]: Maybe:
```
Fixed an issue where the route lines didn't apply `RouteLineResources`'s custom scale expressions correctly.
```?

[the remark this answers is not in the document]
Comment (LukasPaczos) [reply changelog/unreleased/bugfixes/7132.md]: LGTM

**Comment by cafesilencio** [comment]
> We don't have an "inactive leg casing colour" option. It was using travelled colour instead. But I now see that we use travelled casing colour in updateTraveledRouteLine for inactive legs. Is this expected?

Yes it's expected.  The feature request came from Straightaway (and an external developer). As I recall the use case was to make the inactive legs either transparent or nearly transparent by using a color that lessened the visibility.  In that context only one color was needed for styling the inactive legs differently than the active leg.  

The QA app. has 3 activities related to the styling of the inactive route legs.

**Inactive Route Leg Styling:**  In this activity the vanishing route line is set to false. So when the second leg becomes active the first leg takes on the coloring for the inactive leg.

**Inactive Route Leg Styling With Restrictions** In this activity the vanishing route line is set to true so when the second leg becomes active the first leg is not visible (different than the previous activity) because the traveled route line color is transparent.  The traveled color overrides the inactive leg color.

**Active route leg above other legs** This activity was added when the masking layers were added and is probably the most important.  Notice the difference in appearance between the different combinations of options.  Especially with the options:
Unchecked - Vanishing Route Line Transparent
Checked - Enable Vanishing Route Line
Checked - Style Route Legs Differently

This combination will show a yellow inactive route leg and visible traveling route line with the active leg appearing above the inactive legs.  Notice also the appearance of the two legs when there is a transition from the first leg to the second leg. 

Getting the appearance correct with the possible variations in options introduced a lot of complexity.

**Comment by dzinad** [comment]
But why do inactive legs have traveled casing colour, not default casing colour? The route line colour is not the same as traveled route line colour and there is no option to set an inactive leg casing colour. It is always bound to be travaeled casing colour, which isn't obvious from the option name.

**Comment by cafesilencio** [comment]
That could be an oversight.  The alternatively styled inactive leg feature was requested for logistics use cases long before *the masking layers were introduced.  Most customers that use the vanishing route line feature use the default color which is transparent.  So with the casing as transparent by default it could be that what you're pointing out wasn't noticed and might need to be adjusted.  However if you make an adjustment be mindful of the masking layers. 

Additional complexity was added with the requirement that the active leg appear above the inactive legs.  For multi-leg routes with alternate leg styling turned off the inactive legs need to appear like the normal route line below the active leg and for multi-leg routes with alternate styling turned on the inactive legs need to have the alternate coloring and appear below the active leg.

* the masking layers were introduced recently in order to make the active leg appear above the inactive legs

**Pull request merged**